src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/repository/NoteRepository.kt

```diff
@@ -1,16 +1,23 @@
 package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.repository
 
 import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Modifying
 import org.springframework.stereotype.Repository
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.entity.note.EntityType
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.entity.note.NoteEntity
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.entity.note.NoteType
 import java.util.UUID
 
 @Repository
 interface NoteRepository : JpaRepository<NoteEntity, UUID> {
   fun findAllByPrisonNumber(prisonNumber: String): List<NoteEntity>
 
-  fun findAllByEntityReferenceAndEntityType(entityReference: UUID, entityType: EntityType): List<NoteEntity>
+  fun findAllByEntityReferenceAndEntityTypeAndNoteType(entityReference: UUID, entityType: EntityType, noteType: NoteType): List<NoteEntity>
 
   fun findByReference(entityReference: UUID): NoteEntity
+
+  fun findAllByEntityReference(entityReference: UUID): NoteEntity
+
+  @Modifying
```

Comment:
Minor, but is `@Modifying` meant to be used with a `@Query` annotation?

From our old friend [bealdung](https://www.baeldung.com/spring-data-jpa-modifying-annotation):
```
We should note that executing a delete query with @Query works differently from Spring Data JPA’s deleteBy name-derived query methods. The latter first fetches the entities from the database and then deletes them one by one. This means that the life-cycle method @PreRemove will be called on those entities. However, with the former, a single query is executed against the database.
```

With that in mind is it worth adding the `@Query` annotation? It would certainly suggest a small performance improvement in that it will delete the record(s) in one operation, rather than n+1 ? 
(I know we are only deleting 1 note at a time by this method, so any gains will be negligible )

Reply:
Force of habit putting @Modifiying on updates. I'll change it to a query though since that's what the Spring elders recommend.